pkg/jobs/hive/hive.go

```diff
@@ -78,13 +78,34 @@ func MonitorDeployStatus(config *rest.Config, clusterName string) error {
 	if err = utils.LogError(err); err != nil {
 		return err
 	}
-	return monitorDeployStatus(client, hiveset, clusterName, ThreeMinuteMonitorTimeout)
+	return monitorClusterStatus(client, hiveset, clusterName, jobType, ThreeMinuteMonitorTimeout)
 }
 
-func monitorDeployStatus(client clientv1.Client, hiveset hiveclient.Interface, clusterName string, monitorAttempts int) error {
+func DestroyClusterDeployment(hiveset hiveclient.Interface, clusterName string) error {
+	klog.V(0).Infof("Deleting Cluster Deployment for %v\n", clusterName)
+
+	cluster, err := hiveset.HiveV1().ClusterDeployments(clusterName).Get(
+		context.TODO(), clusterName, v1.GetOptions{})
+
+	if err != nil {
+		klog.Warning("Could not retreive cluster " + clusterName + " may have already been deleted")
+		return nil
```

Comment:
so this return nil is assuming it's deleted? shall we use `if errors.IsNotFound(err)` for this path?

Reply:
Instead of introducing a new import I just parsed err.Error() for " not found"

Reply:
I changed my mind and used k8serrors.IsNotFound throughout the entire project.